transmart-core-db/grails-app/services/org/transmartproject/db/querytool/QueriesResourceService.groovy

```diff
@@ -162,11 +166,60 @@ class QueriesResourceService implements QueriesResource {
         resultInstance
     }
 
+    QueryResult runDisablingQuery(Long id,
+                         String username) throws InvalidRequestException
+    {
+        QtQueryResultInstance resultInstance = getQueryResultFromId(id)
+
+        if (resultInstance.queryInstance.userId == username) {
+            resultInstance.deleteFlag = "Y"
+        }
+        else{
+            throw new AccessDeniedException("User ${username} has no permission" +
+                    " to disable query result instance with id $id")
+        }
+
+        def newResultInstance = resultInstance.save()
+        if (!newResultInstance) {
+            throw new RuntimeException('Failure disabling resultInstance ' +
+                    'Errors: ' +
+                    resultInstance.errors)
+        }
+
+        resultInstance
+    }
+
     @Override
     QueryResult getQueryResultFromId(Long id) throws NoSuchResourceException {
-        QtQueryResultInstance.get(id) ?:
-                {  throw new NoSuchResourceException(
-                        "Could not find query result instance with id $id") }()
+        List answer = QtQueryResultInstance.executeQuery(
```

Comment:
Isn't there a GORM-way of doing this instead of writing HQL, like `QtQueryResultInstance.findAllByIdAndDeleteFlagFalse(id)`?